### ISSUE TITLE
Crashed When Exiting the Nether (Ticking Player) (java.nio.file.FileSystemException)

### ISSUE THREAD

**Issue by Coraline77447** [YeetmanLord/So-Many-Enchants#7]
Minecraft: 1.18.1
Forge: 39.1.2
So Many Enchants: 1.1.1

Crash Report: https://pastebin.com/bspx9P5Q
Latest Log: [latest.log](https://github.com/YeetmanLord/So-Many-Enchants/files/8358972/latest.log)
 
Additional Information; 

I think I know what might have happened when I was exiting the Nether into the Overworld..
I have a mod called Bo's Player Companions installed and another mod Called Powder Power, one of my tamed companions is wearing enchanted redium armor and the armor may have some of your enchantments applied onto that armor.
Before I left the nether I'd put my companions back into their companion items that you get after you tame them and I'd put them in my Traveler's Backpack and then I stepped into the portal and the next thing that happens is the game crashes because a file is already in use even tho the game requested the file again for no reason that I know of...
But why didn't the game crashed while I was going into the Nether from the Overworld but it crashes while I'm leaving the Nether to the Overworld?....

**Comment by max-widmaier** [comment]
So this is an error with synching data between server and player so that is not the issue. I think I know why it is happening. Sit tight it is not a compatibility error. I will try to fix this in my next update. Thank you very much for the report

**Comment by max-widmaier** [comment]
If you want technical details basically what is happening is my mod tries to write some data but it can't because the file is already being used by my mod so it just fails and crashes.

**Comment by Coraline77447** [comment]
Oh, ok. Btw I was able to load back in and the last known save was in the Nether before I'd put my companions away before entering the portal. 
The armor that Alex (my rooster from Player Companions mod) was wearing Redium armor from Powder Power mod and it was enchanted with vanilla enchants but I still took off the armor from Alex just in case.
He actually went through the portal without me realizing and was searching for him around the portal in the Nether and finally decided to go through the portal after him and the game didn't crash again.
Alex was standing in the portal waiting for me in the Overworld.

While I was in the Nether with my companions which they were wearing Redium armor which is supposed to provide fire resistance but they were buring alive despite the fact that they were wearing Redium armor that's supposed to protect them from fire and lava damage but thankfully they didn't die from lava (thanks to Marina the Fairy from the same mod) but now I know that the companions from Player Companions can only wear armor that doesn't give status effects like fire resistance, resistance, etc. Like the vanilla armor sets doesn't do any of that..
Only the player is affected when wearing any of Powder Power's armor sets sadly...

I know that what I've said isn't that important to you nor anybody else but I'm just telling yall that my game didn't crash again like how I'd anticipated/expected it too even after taking off the enchanted armor from Alex the Rooster since So Many Enchants affects the vanilla enchants as well.

Speaking of enchants, Enhanced Unbreaking doesn't seem to be working for me at all because the tools durability still decreases like normal like as if I don't have Unbreaking enchantment applied to the items/tools/weapons, so I'd decided to disable Enhanced Unbreaking. I'll only apply Unbreaking on all of my gear instead of Enhanced Unbreaking for now till the issue with that custom enchantment gets fixed.....

**Comment by max-widmaier** [comment]
Hmmm that's weird

**Comment by max-widmaier** [comment]
Latest update should fix this issue!

**Comment by Coraline77447** [comment]
"Hmm that's wierd"
Ikr? I'd thought that Enhanced Unbreaking was working, I had Enhanced Unbreaking on a bow the durability was pretty high but after like 20-30 minutes it's durability was under half already like as if I don't have any unbreaking enchantment on it at all.
I was using the bow periodically but the durability shouldn't have been under half durability already, it should've at least been like 75%-65% left since I think it's durability was at least close to 85% or so but the bow wasn't at full durability (practically new) however. 

Maybe you could test it out for yourself with a new bow or sword or any tool and put the enchantment on any of them and see if Enhanced Unbreaking is working as intended at any of it's levels from 1 to max level. 
Sure I could do that for you and provide you the info on here but I really don't want to go to the config and re-enable it for testing proposes only only to disable it again since I know that Enhanced Unbreaking isn't working as intended..

"latest update should fix this issue!" Okay, so now I'll re-enable Enhanced Unbreaking and see if it's working this time around so that my gear will last longer than regular unbreaking, even tho at higher levels of unbreaking beyond vanilla limit it's pretty good as well but I would imagine that enhanced unbreaking at higher levels than 3 will be better though.
Thanks for fixing the issue(s) btw!

**Comment by max-widmaier** [comment]
Oh dang sorry I forgot about the unbreaking thing I will test that and fix that today

**Comment by Coraline77447** [comment]
WAIT.... Enhanced Unbreaking isn't from your mod, its actually from another enchantment mod called Tons of Enchants.... 
Btw my game crashed yet again from your mod when I was entering the nether this time to go find a nether fortress for wither skeletons....
I'm so sorry for providing false information of an issue with a different mod's enchantment, that custom enchantment isn't from your mod but why I'd thought that it was was cause both So Many Enchants and Tons of Enchants sounds alike and got the enchantments as of which mod they're from all mixed up by accident..
So you didn't forget anything regarding your unbreaking enchantment changes...
How I'd finally figured all of this mess out was when your mod crashed my game again while entering the nether and told myself "I'll have to uninstall So Many Enchants since it's breaking my game! At least I still have Tons of Enchants to play with" or something like that
Anyways I'm uninstalling So Many Enchants till this issue is fixed.. 
Also what the latest log tells me now is that the mod/game is trying to save a closed config file or something like that?...

When I looked in the config folder I see multiple config files of your mod but most of the files aren't a TOML file, they're BAK files but there's one that's the real config file for the mod.

[latest.log](https://github.com/YeetmanLord/So-Many-Enchants/files/8380978/latest.log)

So the problem may not have been fixed after all since the game crashed while entering the nether....

**Comment by max-widmaier** [comment]
Bruh ok thanks

**Comment by max-widmaier** [comment]
Ok the latest update _should_ fix the crash

**Comment by Coraline77447** [comment]
Well lets hope that it is but sadly I've already uninstalled So Many Enchants along with Tons of Enchants for one of my two 1.18.1 modded worlds (world name: Cora Tinkers')..
Before I'd uninstalled both mods I had to get rid of all of those enchantments and had to make/rebuild my Gemium gear (from Powder Power mod btw) that had some of your enchantments along with Tons of Enchants on them as well as getting rid of some of the armors, tools and weapons that were enchanted with your enchantments and Tons of Enchants's enchantments... 
Both mods were replaced by MarbleGate's Exotic Enchantment: Flowing Agony.

Unfortunately some or all of the weaponsmith, toolsmith, and armor villagers in all of the villages that I've found them in inclouding my cured villager who's a toolsmith in Cora Tinkers' still have most of their trades all messed up since most of their enchanted tools (iron & diamond), weapons (iron & diamond), and diamond armor trade will have blank enchantments since they used to be enchanted with yours and Tons of Enchants's enchantments and there's no real good way to reset them without killing them...
-Either killing them by; the player, using /kill, even tho that'll kill everything around the player as well as the player. 
-Or letting the zombies kill/eat them and hope that they turn into villager zombies to cure them back to villagers to reset their trades if they decide to be who they once were before their death from zombies but if they decide not to be who they once were then you've lost their professions entirely forever....

On my other world that I'm going onto today as well as tomorrow and the day after that I'll have to get rid of Tons of Enchants's enchantments since one of their enchantments isn't worling as intended (Enhanced Unbreaking) but at least I don't need to get rid of your enchantments since I'll be updating your mod in that world's mods list (world name: Cora World Redo).
Also that world already has Flowing Agony installed as well.

Again, thx for fixing the crashes and again lets hope that they're finally fixed this time around.

-*I typically don't really need to do any of that nor even trade with them since I still have the enchanting table and a butt load of XP (only on Cora Tinkers' sadly) stored in the Global XP block from Global XP mod.
(I do have that mod installed on the other world as well but it doesn't have as much XP stored in it as the one on Cora Tinkers' since in Cora Tinkers' I was doing mob killing sprees and gained alot of levels doing it as well as Powder Power's Experience Token which I don't have that item in the other world yet since it's kinda expensive to craft with some rare ingredients (Bottle O' Experience))*-
